specifyweb/businessrules/attachment_rules.py

```diff
@@ -42,7 +42,8 @@ def attachment_save(attachment):
 @orm_signal_handler('post_delete', 'Attachment')
 def attachment_deletion(attachment):
     from specifyweb.attachment_gw.views import delete_attachment_file
-    delete_attachment_file(attachment.attachmentlocation)
+    if attachment.attachmentlocation is not None:
+        delete_attachment_file(attachment.attachmentlocation)
```

Comment:
The only concern with these changes is if Specify was able to delete (or try and delete) an attachment file when its `attachmentlocation` is None. 
Of course, this does not make sense logically, but it is the only implication of this change to consider. 
I have gone through the code briskly and am pretty sure this could not be the case before. 

https://github.com/specify/specify7/blob/80b2cd818408672e0b0f61d19125f5bbc5ab7ce0/specifyweb/attachment_gw/views.py#L165-L176

Ultimately, it comes down to how the Asset Server handles null values in the body of the `filedelete` endpoint.

Reply:
Actually, a None attachmentlocation would throw an error in Specify currently anyways, stating something along the lines of: "NoneType object has no attribute `encode`"

https://github.com/specify/specify7/blob/80b2cd818408672e0b0f61d19125f5bbc5ab7ce0/specifyweb/attachment_gw/views.py#L183-L185

(I'll leave this as a comment anyways, for documentation purposes)

Reply:
Yep, that's why I used attachmentlocation!